src/transformers/configuration_prophetnet.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+# Copyright 2020 The Microsoft Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" ProphetNet model configuration """
+
+
+import logging
+
+from .configuration_utils import PretrainedConfig
+
+
+logger = logging.getLogger(__name__)
+
+PROPHETNET_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "microsoft/prophetnet-large-uncased": "https://s3.amazonaws.com/models.huggingface.co/bert/microsoft/prophetnet-large-uncased/config.json",
+    "microsoft/xprophetnet-large-wiki100-cased": "https://s3.amazonaws.com/models.huggingface.co/bert/microsoft/xprophetnet-large-wiki100-cased/config.json",
+}
+
+
+class ProphetNetConfig(PretrainedConfig):
+    r"""
+            Configuration class for Bart. Parameters are renamed from the fairseq implementation
+        """
+    model_type = "prophetnet"
+
+    def __init__(
+            self,
+            activation_dropout=0.1,
+            activation_function="gelu",
+            vocab_size=30522,
+            d_model=24,
+            encoder_ffn_dim=96,
+            encoder_layers=2,
+            encoder_attention_heads=2,
+            decoder_ffn_dim=96,
+            decoder_layers=2,
+            decoder_attention_heads=2,
+            encoder_layerdrop=0.0,
+            decoder_layerdrop=0.0,
+            attention_dropout=0.0,
+            dropout=0.1,
+            max_position_embeddings=1024,
+            init_std=0.02,
+            num_labels=3,
+            is_encoder_decoder=True,
+            pad_token_id=0,
+            bos_token_id=102,
+            eos_token_id=102,
+            ngram=2,
+            num_buckets=32,
+            relative_max_distance=128,
+            disable_ngram_loss=False,
+            eps=0.0,
+            **common_kwargs
+    ):
+        if "hidden_size" in common_kwargs:
```

Comment:
Do we have to call the parameter `d_model`? It would be great if we could only use `hidden_size` to avoid confusion. While a lot of previous models have `d_model` we are trying to be more consistent now with the naming.